torch/csrc/distributed/rpc/BuiltinRet.cpp

```diff
@@ -0,0 +1,43 @@
+#include <torch/csrc/distributed/rpc/BuiltinRet.h>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+BuiltinRet::BuiltinRet(std::vector<at::IValue>&& values) : values_(values) {}
+
+BuiltinRet::~BuiltinRet() = default;
+
+std::vector<at::IValue>& BuiltinRet::values() {
+  return values_;
+}
+
+Message BuiltinRet::toMessage() {
+  std::vector<torch::Tensor> tensor_table;
+  Pickler pickler(&tensor_table);
+
+  pickler.protocol();
+  pickler.startTuple();
+  for (auto& value: values_) {
+    pickler.pushIValue(value);
+  }
+  pickler.endTuple();
```

Comment:
Hmm should this really do the tuple wrapping here instead of letting the function handle it? No user-defined functions will have multiple returns, and this will make a two-level tuple unnecessarily

Reply:
Yep, I will use the new pickler API when #23241 is merged. Some recent changes in `Unpickler` disallow serializing multiple `IValue`s in the stack due to [this](https://github.com/pytorch/pytorch/blob/c09e92255c69828878e7ea851eccb0699b6af2c0/torch/csrc/jit/pickler.cpp#L491-L494), so I packed them into one `IValue`.

Reply:
confirmed with @driazati, quote:

> We'd want everything to match Python's pickle.dump() which only takes a single value, so everything would have to be wrapped in a tuple by the caller